### DISCUSSION TITLE
Benchmark question

### DISCUSSION THREAD

**Discussion by Kludex** [cirospaciari/socketify.py#56] [Q&A] [answered]
Hi there,

Do you perhaps have the packages installed on the benchmark you performed?

**Comment by cirospaciari** [comment, marked as answer]
@Kludex good question!
You can check in https://github.com/TechEmpower/FrameworkBenchmarks the packages in docker for each benchmark.

In the case of WebSockets:
Runtime versions: PyPy3 7.3.9, Python 3.10.7, node v16.17.0, bun v0.2.2
Framework versions: gunicorn 20.1.0 + uvicorn 0.19.0, socketify beta-4.1, gunicorn 20.1.0 + falcon 3.1.0

I you need any help replicating any benchmarks, just reach me here or in the discord https://discord.socketify.dev/ will be a pleasure

**Reply by cirospaciari** [reply]
In case of TFB plaintext you can see some old results in: https://www.techempower.com/benchmarks/#section=test&runid=124fd2af-3030-4315-8876-e1db1fa91193&test=plaintext&l=hra0hr-35r

And in future and new rounds you will be able to see socketify WSGI and ASGI servers in techempower benchmarks site too.
https://tfb-status.techempower.com/

**Reply by cirospaciari** [reply]
Acabei de ver que é brasileiro tambem haha, qualquer coisa pode falar em português tambem :brazil:

**Reply by Kludex** [reply]
https://github.com/TechEmpower/FrameworkBenchmarks/pull/7613
https://github.com/TechEmpower/FrameworkBenchmarks/pull/7808

:shrug:

**Reply by cirospaciari** [reply]
> [TechEmpower/FrameworkBenchmarks#7613](https://github.com/TechEmpower/FrameworkBenchmarks/pull/7613) [TechEmpower/FrameworkBenchmarks#7808](https://github.com/TechEmpower/FrameworkBenchmarks/pull/7808)
> 
> shrug

@Kludex I did not know about this issue, I will re-test here and update the numbers.

**Reply by cirospaciari** [reply]
Just a quick update fastsapi:

with:
uvloop==0.17.0
uvicorn==0.20.0
httptools==0.5.0

37k fastapi uvicorn asgi python3

with the current version in TFB:
16k fastapi uvicorn asgi python3

~I will add it to emmet uvicorn tests and falcon asgi tests too~ Actually emmett already adds these dependencies, running to check falcon and I will re-run the websockets benchmarks too.
